velox/exec/tests/RuntimeErrorMsgTest.cpp

```diff
@@ -0,0 +1,167 @@
+    FAIL() << "Expected a MEM_CAP_EXCEEDED RuntimeException. Plan Stats: "
+           << printPlanWithStats(*plan, task->taskStats());
+  } catch (const VeloxRuntimeError& e) {
+    ASSERT_STREQ(expectedErrorMsg.c_str(), e.message().c_str());
+    ASSERT_STREQ("MEM_CAP_EXCEEDED", e.errorCode().c_str());
+  } catch (const std::exception& e) {
+    FAIL() << "Expected a MEM_CAP_EXCEEDED RuntimeException but received"
+           << e.what();
+  }
+}
+
+TEST_F(RuntimeErrorMsgTest, multipleDriversMemCapExceeded) {
+  // Executes a plan that runs with two drivers and query memory limit that
+  // forces it to throw MEM_CAP_EXCEEDED exception. Verifies that the Top 3
+  // Operators by memory usage list in the error message contains operators from
+  // multiple drivers. This is verified by confirming that there are multiple
+  // driver ids in the error message. A small plan and only two drivers are used
+  // to ensure that the test deterministically includes two driver ids in its
+  // error message.
+  vector_size_t size = 1'024;
+  const int32_t numSplits = 100;
+  constexpr int64_t kMaxBytes = 3LL << 20; // 3MB
+  std::vector<RowVectorPtr> data;
+  auto dataFiles = makeFilePaths(numSplits);
+  for (auto i = 0; i < numSplits; ++i) {
+    auto rowVector = makeRowVector({
+        makeFlatVector<int32_t>(
+            size, [&i](auto row) { return row + (i * 1000); }),
+        makeFlatVector<int32_t>(size, [](auto row) { return row + 3; }),
+    });
+    data.push_back(rowVector);
+    writeToFile(dataFiles[i]->path, rowVector);
+  }
+
+  createDuckDbTable("tmp", data);
+  auto rowType = std::dynamic_pointer_cast<const RowType>(data[0]->type());
+  core::PlanNodeId scanNodeId;
+  auto plan = PlanBuilder()
+                  .tableScan(rowType)
+                  .capturePlanNodeId(scanNodeId)
+                  .singleAggregation({"c0"}, {"sum(c1)"})
+                  .planNode();
+  auto queryCtx = core::QueryCtx::createForTest();
+  queryCtx->pool()->setMemoryUsageTracker(
+      velox::memory::MemoryUsageTracker::create(
+          kMaxBytes, kMaxBytes, kMaxBytes));
+
+  std::string errorMsg;
+  try {
+    auto task =
+        AssertQueryBuilder(plan, duckDbQueryRunner_)
+            .queryCtx(queryCtx)
+            .maxDrivers(2)
+            .splits(scanNodeId, makeHiveConnectorSplits(dataFiles))
+            .assertResults("SELECT c0, sum(c0 + 2) FROM tmp GROUP BY 1");
+    FAIL() << "Expected a MEM_CAP_EXCEEDED RuntimeException. Plan Stats: "
+           << printPlanWithStats(*plan, task->taskStats());
+  } catch (const VeloxRuntimeError& e) {
+    ASSERT_STREQ("MEM_CAP_EXCEEDED", e.errorCode().c_str());
+    errorMsg = e.message();
+  } catch (const std::exception& e) {
+    FAIL() << "Expected a MEM_CAP_EXCEEDED RuntimeException but received"
+           << e.what();
+  }
+
+  // Verify multiple driver ids are a part of the error message
+  const std::regex operatorIdRegex("operatorId: (\\d+),");
```

Comment:
Driver IDs are not user-friendly. I wonder if in production scenario the same operator will show up in the list of top 3. In this case, it might be helpful to include such operator once, but maybe specify that it hit the memory limit in N drivers.